core/state/database.go

```diff
@@ -141,6 +145,50 @@ type Trie interface {
 	IsVerkle() bool
 }
 
+// TransitionState is a structure that holds the progress markers of the
+// translation process.
+type TransitionState struct {
+	CurrentAccountAddress *common.Address // addresss of the last translated account
```

Comment:
Using pointer is a bit over-complicated, we can replace it with the account address hash, reasons:

- the verkle migration is traversed with the order of account address hash;
- the zero hash can represent nothing has been migrated yet
- using hash can make the entire TransitionState as a pure value, no pointer

Using a pointer here is unnecessarily complex, we can replace it with the account address hash instead, reasons:
- Verkle migration is traversed in the order of account address hashes
- A zero hash can represent that nothing has been migrated yet
- Using a hash allows the entire TransitionState to remain a pure value type, with no pointers involved

Reply:
> I could confirm that not using the pointer was breaking the changes owing to some side effect somewhere else that I don't remember just now

Yeah, maybe I miss something. I can live with the current workaround. But I will always keep my suggestions to simplify the code.